src/tracing/perfetto/perfetto_tracing.h

```diff
@@ -38,12 +40,16 @@ class PerfettoBackend : public ::chip::Tracing::Backend
     // as they would be slower than expected. Perfetto trace macros
     // are expected to be set exclusively (via matter_trace_config)
 
+    void TraceCounter(const char * label, const char * group) override;
     void LogMessageSend(MessageSendInfo &) override;
     void LogMessageReceived(MessageReceivedInfo &) override;
 
     void LogNodeLookup(NodeLookupInfo &) override;
     void LogNodeDiscovered(NodeDiscoveredInfo &) override;
     void LogNodeDiscoveryFailed(NodeDiscoveryFailedInfo &) override;
+
+private:
+    std::unordered_map<std::string, int> mCounters;
```

Comment:
perfetto has its own counter macros: https://perfetto.dev/docs/instrumentation/track-events#counters

we should not re-invent them.

Reply:
The logic in perfetto is that it does not allow multiplexing, so events, scopes and counters will go via macros_impl only - see comment above. We expect these to be set exclusively.

This is because perfetto did a lot of work on performance tuning so we do not want to bypass that.